packages/core-components/src/layout/Sidebar/SidebarGroup.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BackstageTheme } from '@backstage/theme';
+import {
+  BottomNavigationAction,
+  BottomNavigationActionProps,
+  makeStyles,
+  useMediaQuery,
+} from '@material-ui/core';
+import React, { useContext } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Link } from '../../components';
+import { sidebarConfig } from './config';
+import { MobileSidebarContext } from './MobileSidebar';
+
+interface SidebarGroupProps extends BottomNavigationActionProps {
+  to?: string;
+  priority?: number;
+}
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  root: {
+    flexGrow: 0,
+    margin: `0 ${theme.spacing(2)}px`,
+    color: theme.palette.navigation.color,
+  },
+
+  selected: {
+    color: `${theme.palette.navigation.selectedColor}!important`,
+    borderTop: `solid ${sidebarConfig.selectedIndicatorWidth}px ${theme.palette.navigation.indicator}`,
+    marginTop: '-1px',
+  },
+
+  label: {
+    display: 'none',
+  },
+}));
+
+export const SidebarGroup = ({
+  to,
+  label,
+  icon,
+  value,
+  children,
+}: React.PropsWithChildren<SidebarGroupProps>) => {
+  const classes = useStyles();
+  const location = useLocation();
+  const { selectedMenuItemIndex, setSelectedMenuItemIndex } =
```

Comment:
Feeling that we should take a bunch of this logic and move it all into an internal `MobileSidebarGroup` component, just to keep mobile vs desktop separated in this implementation